src/attributes.rs

```diff
@@ -101,6 +102,36 @@ let value: Vec<u8> = value.try_into().unwrap();
 assert_eq!(data, value);
 ```
 
+# magic
+
+Sets a "magic" value that must be present in the data at the start of
+a struct/enum when reading, and that is written out of the start of
+that type's data when writing, without needing to be explicitly stored
+in a field
+
+Example:
+```rust
+# use deku::prelude::*;
+# use std::convert::{TryInto, TryFrom};
+# #[derive(Debug, PartialEq, DekuRead, DekuWrite)]
+#[deku(magic = b"deku")]
+pub struct MagicDeku {
```

Comment:
nit: `DekuTest` is used as the struct names in the docs for consistency between examples